### PR TITLE
Update mutation for adding Household members

### PR DESCRIPTION
**Pull request by gigxz**
* Update mutation so that it can be also be used for _creating_ a new household. This supports the new household creation from within a project context.
* Remove code for adding multiple members at a time, that's not a thing we're doing or planning to do
* Raise errors instead of returning validations in more cases

Screen Shot 2023-06-18 at 3 27 17 PM

### CODE REVIEW

**Review by ttoomey — APPROVED** [review]
Looks good to me!

File: drivers/hmis/app/graphql/mutations/add_to_household.rb [hunk 1]
```diff
@@ -0,0 +1,71 @@
...
+        raise HmisErrors::ApiError, 'Household ID not found' unless existing_enrollments.exists?
+        raise HmisErrors::ApiError, 'Access denied' unless current_user.permissions_for?(project, :can_edit_enrollments)
+        raise HmisErrors::ApiError, 'Mismatched Project ID' if existing_enrollments.first.project.id != project.id
+        raise HmisErrors::ApiError, 'Client is already a member of this household' if existing_enrollments.joins(:client).where(c_t[:id].eq(client_id)).exists?
+
+        errors.add :relationship_to_hoh, :invalid, full_message: 'Household already has a Head of Household' if is_hoh && existing_enrollments.heads_of_households.exists?
+      else
+        raise HmisErrors::ApiError, 'Access denied' unless current_user.permissions_for?(project, :can_enroll_clients)
+
+        errors.add :relationship_to_hoh, :invalid, full_message: 'Client must be the Head of Household' unless is_hoh
```
Comment (eanders) [inline drivers/hmis/app/graphql/mutations/add_to_household.rb]: Do we want to raise this, or just set it to `1`?

**Review by eanders — APPROVED** [review]
Looks good to me, just one question about if we want to raise an error when we don't have an HoH and are adding the first person in the household, or if we should just make them the HoH.

File: drivers/hmis/app/graphql/mutations/add_to_household.rb (hunk 1 above)
Comment (gigxz) [reply drivers/hmis/app/graphql/mutations/add_to_household.rb]: Yeah I think we should probably just make them the HoH. We do that on the frontend already, so I don't expect this to happen, but it seems safer. I'll update it.

**Comment by gigxz** [comment]
A test was failing with an access issue that I couldn't reproduce locally, but I think is related to the way the perm setup in tests is not quite right (there is a `TODO` to set up for ACLs). I added a new approach https://github.com/greenriver/hmis-warehouse/pull/3176/commits/7344031fd3d3457d239a18ea66e3576a43414e76 and will do a separate PR to update other tests accordingly.

**Comment by gigxz** [comment]
https://github.com/greenriver/hmis-warehouse/pull/3178 updates tests to use new ACL approach

**Pull request merged**